enterprise/app/trends/heatmap.tsx

```diff
@@ -524,17 +518,24 @@ class HeatmapComponentInternal extends React.Component<HeatmapProps, State> {
   }
 
   render() {
-    const width = this.getWidth() - CHART_MARGINS.left - CHART_MARGINS.right;
-    const height = this.getHeight() - CHART_MARGINS.top - CHART_MARGINS.bottom;
+    const { width, ref } = useResizeDetector({
+      handleHeight: false,
+      refreshMode: "throttle",
+      refreshRate: 500,
+    });
```

Comment:
does this work? I thought React hooks (functions prefixed with `use`) only worked with functional components, and weren't supported for class components

Reply:
Then you're probably right; I couldn't test this locally and so was going to test it in dev when I got a free moment. How do you recommend we resolve this? The library just removed the `with` version, and I wasn't aware of the differing use in class and functional components until just now.

Reply:
Personally I'd ignore the deprecation warning

Reply:
(and revert to the old version - unless the issue is that this is setting off some security scanner?)

Reply:
I switched the `HeatmapComponent` to a functional component instead of a class component, and it appears to work based on my limited testing I did on my local machine with no real data.